### ISSUE TITLE
WPF Plot Stops Refreshing Until Mouse Panning Stops

### ISSUE THREAD

**Issue by jbuckmccready** [ScottPlot/ScottPlot#1387]
## Bug Report

**Issue:** After a brief time of continuous mouse panning (a second or so) the plot stops refreshing until the mouse stops moving.

**Reproducing:** Click and drag pan constantly (do not let the mouse stop), at first everything refreshes and works but then it stops refreshing and as long as the mouse is kept moving it never redraws, once the mouse stops (even with mouse button not released) it refreshes but has the problem again if panning starts again. The bug is consistently reproducible with a debugger attached to essentially an empty example (see code, axes stop refreshing), or when using the exact same example code in a larger application (without debugger attached). I am not sure why it doesn't happen in all cases. I have investigated and tried toggling/changing all of the configuration quality options (under `WpfPlot.Configuration.Quality` and `WpfPlot.Configuration.QualityConfiguration`) but nothing seems to fix the issue.

This issue is also always reproducible when running the WPF ScottPlot Demo or sandbox/WpfApp in the repository with debugger attached (but does not have the problem when running without debugger). This issue does not seem to be reproducible in WinForms with debugger attached.

I don't mind the problem being present with a debugger attached, but I have the problem without a debugger attached when using ScottPlot.WPF in a larger application.

```cs
public partial class MainWindow : Window
{
    public MainWindow()
    {
        InitializeComponent();
        WpfPlot.Plot.Clear();
        WpfPlot.Refresh();
    }
}
```

```xaml
<Window x:Class="GeometryScene.MainWindow"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d"
        Title="MainWindow" Height="450" Width="800">
    <Grid x:Name="MainGrid">
        <WpfPlot x:Name="WpfPlot"></WpfPlot>
    </Grid>
</Window>

```

### System Details
* **ScottPlot Version:** 4.1.26
* **Operating System:** Windows 10
* **Application Type:** WPF
* **.NET Version:** NET Framework 4.7.1 or NET Framework 4.8.1

**Comment by jbuckmccready** [comment]
Setting `WpfPlot.Configuration.UseRenderQueue = true` seems to help but it has the same problem just less frequently.

**Comment by jbuckmccready** [comment]
I fixed the issue (and improved the performance of the interactive WPF rendering noticeably on my machine) by using a WriteableBitmap which avoids some unneeded allocations and rebinding the WPF `Image.Source` property when panning.

Please let me know if there's anything else I need to do for the pull request, thanks for the great library!

**Comment by swharden** [comment]
Thank you for fixing this @jbuckmccready! It was on my to-do list to try to and fix this tonight. I'm not sure where this behavior crept in (and whether or not it only affects my .NET 6 preview builds) but it's not good.

I'll probably swap `UseRenderQueue` to `true` by default, and in combination with #1388 hopefully things will be back to normal. I'll take a closer look this evening 👍

**Comment by riquich** [comment]
I just tried and want to update on this. 

Merging the pull resolves pan-stop issue but I believe the merge changes something related with DPI. I use custom visual layers on top of original ScottPlot image, which depends on ScottPlot's coordinate functions. After the update visuals in my visual layers are misplaced to wrong locations. Like 25% off or so (I'm using 125% DPI), and the ScottPlot image seems abit blurry. So deliberating on the effects of the merge abit more might be a safe option

**Comment by swharden** [comment]
I can take a closer look at this later, but in case it wasn't mentioned above note that an option to enable/disable display scale stretching was very recently added (it's in the master branch but not on NuGet yet) https://github.com/ScottPlot/ScottPlot/pull/1364

Using that new option may either improve this situation, or if it introduced undesirable behavior we could look closely at what was merged to have a better idea of what may be changed

**Comment by riquich** [comment]
now that I tried again with and without PR, it seems both breaks my current DPI related works. So you are right, it is not about the PR. The changes seems to be stemming from the differences between current nuget release and latest source code. Thanks for investigating

**Comment by swharden** [comment]
@riquich this issue was closed because I just merged #1388 but I wanted to follow-up on this...

> @riquich: After the update visuals in my visual layers are misplaced to wrong locations. Like 25% off or so (I'm using 125% DPI), and the ScottPlot image seems abit blurry, and the axis label texts become bigger. So deliberating on the effects of the merge abit more might be a safe option

> @riquich: The changes seems to be stemming from the differences between current nuget release and latest source code (#1364)

Hi @riquich, thanks for calling this out! I'm surprised to learn you're experiencing unexpected behavior because I intended #1364 to have no change in default behavior. That PR has been merged to the main branch and I was planning to push it to NuGet tonight, but your comment makes me want to double-check what may be going on here.

The latest source code in the main branch has a DPI scaling demo at the bottom of the demo application (WPF demo, not the WinForms demo). Is the default behavior still surprising to you, and/or can you meet your program's goals by setting the `DpiStretch` configuration option?

```cs
WpfPlot1.Configuration.DpiStretch = true; // scaled but blurry
WpfPlot1.Configuration.DpiStretch = false; // 1:1 pixel resolution but small
```

If your answer is advanced it may be worth discussing further on #1364 instead of here 👍
